internal/pkg/model/v2/activity/service/ratelimiter.go

```diff
@@ -0,0 +1,146 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/ulule/limiter/drivers/store/memory"
+
+	"github.com/ulule/limiter"
+)
+
+// NewLimiter creates new limiter with specified limit
+func NewLimiter(limit string) *limiter.Limiter {
+	//create rate
+	rate, err := limiter.NewRateFromFormatted(limit)
+	if err != nil {
+		panic(err)
+	}
+	//create store
+	store := memory.NewStore()
+	//create limiter
+	limiter := limiter.New(store, rate)
+	return limiter
+}
+
+// Limiters is set of rate limiters
+type Limiters struct {
+	limiters map[string]*limiter.Limiter
+	sync.RWMutex
+}
+
+// Lookup looks up limiter with specified name
+func (l *Limiters) Lookup(name string, limit string) *limiter.Limiter {
+	l.RLock()
+	limiter := l.limiters[name]
+	l.RUnlock()
+
+	if limiter != nil {
+		return limiter
+	}
+
+	limiter = NewLimiter(limit)
+	l.Lock()
+	l.limiters[name] = limiter
+	l.Unlock()
+
+	return limiter
+}
+
+var limiters = Limiters{
+	limiters: make(map[string]*limiter.Limiter),
+}
+
+// RateLimiter is a rate limiter service
+// Limit can be specified in the format "<limit>-<period>"
+//
+// Valid periods:
+// * "S": second
+// * "M": minute
+// * "H": hour
+//
+// Examples:
+// * 5 requests / second : "5-S"
+// * 5 requests / minute : "5-M"
+// * 5 requests / hour : "5-H"
+type RateLimiter struct {
+	Name string
+
+	// inputs
+	Limit string `json:"limit"`
+	Token string `json:"token"`
+
+	// outputs
+	LimitReached   bool   `json:"limitReached"`
+	LimitAvailable int64  `json:"limitAvailable"`
+	Error          bool   `json:"error"`
+	ErrorMessage   string `json:"errorMessage"`
+}
+
+// Execute invokes this service
+func (rl *RateLimiter) Execute() (err error) {
+	// check for request token
+	if rl.Token == "" {
```

Comment:
Would it make sense to have a sensible default if a token is not provided? We could assume no token means the rate limiter is global and perhaps use "Global" as the token string?

Reply:
@jeffreybozek No token case is not handled here as I want to give flexibility to the user whether to operate it as global rate limiter or not.
I have improved docs section to call this out - https://github.com/TIBCOSoftware/mashling/tree/feature-basic-ratelimiter-service/docs/gateway#services-rate-limiter -> *Note:* section.

Also added an example under examples/recipes/v2 floder.